packages/dm-core-plugins/src/form/components/AttributeList.tsx

```diff
@@ -35,10 +43,18 @@ export const AttributeList = (props: {
           namePath={`${prefix}${attribute.name}`}
           attribute={attribute}
           uiAttribute={uiAttribute}
+          readOnly={readOnly}
         />
       </div>
     )
   })
 
-  return <Stack spacing={1}>{attributeFields}</Stack>
+  return (
+    <Stack spacing={1}>
+      {config?.readOnly && (
```

Comment:
This way, if the config is set to readonly, it will always be possible to edit the form anyway be toggling. Thus it is still impossible to create a form where editing is impossible, which I think is what Babak wanted.

I think that if we want a toggle button or not should be defined by yet another config setting. And then perhaps rename config.readonly to config.initiallyReadOnly.

Reply:
Agreed!